src/util.ts

```diff
@@ -1133,6 +1142,9 @@ export class Util {
                         // the official plugin format is a factory function that returns a new instance of a plugin.
                     } else if (typeof theExport === 'function') {
                         plugin = theExport();
+                    } else {
+                        //this should never happen; somehow an invalid plugin has made it into here
+                        throw new Error(`TILT: Encountered an invalid plugin: ${String(plugin)}`);
```

Comment:
I tend to sprinkle these assertions into my code, but they are often considered unidiomatic. The alternative here would be to remove the `if` condition from the `else if` above. Removing the condition could allow bad data to get farther into the program before it crashes, but so long as the type of `theExport` (`CompilerPlugin | CompilerPluginFactory`) is correct, no `if` is needed in the `else`.